custom_model_runner/datarobot_drum/drum/utils.py

```diff
@@ -121,26 +122,29 @@ def shared_fit_preprocessing(fit_class):
         fit_class.num_rows = int(fit_class.num_rows)
 
     # get target and features, resample and modify nrows if needed
-    if fit_class.target_filename:
-        X = df.sample(fit_class.num_rows, random_state=1)
-        y = pd.read_csv(fit_class.target_filename, index_col=False).sample(
-            fit_class.num_rows, random_state=1
-        )
-        assert len(y.columns) == 1
-        assert len(X) == len(y)
-        y = y.iloc[:, 0]
-    elif fit_class.target_name:
+    if fit_class.target_filename or fit_class.target_name:
+        if fit_class.target_filename:
+            y_unsampled = pd.read_csv(fit_class.target_filename, index_col=False)
+            assert len(y_unsampled.columns) == 1
+            assert len(df) == len(y_unsampled)
+            df = df.merge(y_unsampled, left_index=True, right_index=True)
+            assert len(y_unsampled.columns.values) == 1
+            fit_class.target_name = y_unsampled.columns.values[0]
+        df = df.dropna(subset=[fit_class.target_name])
         X = df.drop(fit_class.target_name, axis=1).sample(
             fit_class.num_rows, random_state=1, replace=True
         )
         y = df[fit_class.target_name].sample(fit_class.num_rows, random_state=1, replace=True)
+
+        if y.dtype == np.dtype("bool"):
```

Comment:
Same, should we manipulate data?

Reply:
This could also migrate into the model templates, you're right.